metals/src/main/scala/scala/meta/internal/metals/DefinitionProvider.scala

```diff
@@ -53,6 +55,30 @@ final class DefinitionProvider(
     }
   }
 
+  def symbolOccurrences(
```

Comment:
Let's create a `DocumentHighlightProvider` class that has `DefinitionProvider` as a primary constructor parameter.